iso15118/secc/states/iso15118_2_states.py

```diff
@@ -765,6 +768,49 @@ class PaymentDetails(StateSECC):
     def __init__(self, comm_session: SECCCommunicationSession):
         super().__init__(comm_session, Timeouts.V2G_SECC_SEQUENCE_TIMEOUT)
 
+    def _all_certificates_from_chain(
+        self, certificate_chain: CertificateChain
+    ) -> List[Certificate]:
+        """Return all certificates from a certificate chain as a list.
+
+        The order should be: leaf certificate, sub-CA 2, sub-CA 1,
+        if all three are present.
+        """
+        return [
+            certificate_chain.certificate
+        ] + certificate_chain.sub_certificates.certificates
+
+    def _get_contract_certificate_hash_data(
+        self,
+        certificate_chain: Optional[CertificateChain],
+    ) -> Optional[List[Dict[str, str]]]:
+        """Return a list of hash data for a contract certificate chain."""
+        if certificate_chain is not None:
+            return None
+
+        return [
+            derive_certificate_hash_data(certificate)
+            for certificate in self._all_certificates_from_chain(certificate_chain)
+        ]
+
+    def _build_pem_certificate_chain(
+        self, certificate_chain: Optional[CertificateChain]
+    ) -> Optional[str]:
+        """Return a string of certificates in PEM form concatenated together."""
+        if certificate_chain is None:
+            return None
+
+        return "".join(
+            [
+                certificate_to_pem_string(certificate)
+                for certificate in self._all_certificates_from_chain(certificate_chain)
+            ]
+        )
```

Comment:
Think it would be good to add some unit tests for each of these methods to verify the behaviour.

Reply:
Done now.